File: dungeons-26.1.2/src/main/java/fishmod/utils/rendering/DrawHandler.java
package fishmod.utils.rendering;

import java.util.ArrayList;
import java.util.List;
import java.util.function.Consumer;

public class DrawHandler <T> {

    private final List<T> listeners = new ArrayList<>();

    public void register(T listener) {
        listeners.add(listener);
    }

    public void invoke(Consumer<T> action) {
        if (listeners.isEmpty()) return;
        for (T listener : listeners) {
            action.accept(listener);
        }
    }

}

File: dungeons-26.1.2/src/main/kotlin/fishmod/utils/rendering/DrawHandler.kt
package fishmod.utils.rendering

import java.util.function.Consumer

class DrawHandler<T> {

    private val listeners = ArrayList<T>()

    fun register(listener: T) {
        listeners.add(listener)
    }

    fun invoke(action: Consumer<T>) {
        if (listeners.isEmpty()) return
        for (listener in listeners) {
            action.accept(listener)
        }
    }

}
